### PR TITLE
Add destructors.

### PR DESCRIPTION
**Pull request by MaxJPRey**
Fix issue #382.
Add a destructor for the application/design's variables.

### CODE REVIEW

File: pyaedt/application/Design.py [hunk 1]
```diff
@@ -516,6 +516,11 @@ def __init__(self, design_type, project_name=None, design_name=None, solution_ty
         self.project_datasets = self._get_project_datasets()
         self.design_datasets = self._get_design_datasets()
 
+        def __del__(self, key):
+            """Implement destructor with array name or index."""
+            object_to_delete = self._variable_manager(key)
+            del object_to_delete
```
Comment (akaszynski) [inline pyaedt/application/Design.py:522]: This should be simplified with
```suggestion
            del self._variable_manager[key]
```

**Comment by akaszynski** [comment]
This also needs a test to verify it works and to ensure we maintain code coverage.

File: pyaedt/application/Design.py (hunk 1 above)
Comment (maxcapodi78) [reply pyaedt/application/Design.py:522]: I agree. we should follow @akaszynski suggestion

**Review by akaszynski — APPROVED** [review]
Good work.  This is ready to merge.

**Pull request merged**